src/main/java/ua/dymohlo/multithreadingcontinue/ProducerConsumerProblemWIthQueue.java

```diff
@@ -0,0 +1,48 @@
+package ua.dymohlo.multithreadingcontinue;
+
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+
+public class ProducerConsumerProblemWIthQueue {
+    public static final BlockingQueue<Integer> buffer = new ArrayBlockingQueue<>(10);
+
+    public static void problemWIthQueue() {
+        Thread producer = new Thread(() -> {
+            while (true) {
+                produce();
+            }
+        });
+
+        Thread consumer = new Thread(() -> {
+            while (true) {
+                consumer();
+            }
+        });
+
+        producer.start();
+        consumer.start();
+    }
+
+    private static void produce() {
+        int randomInt = ThreadLocalRandom.current().nextInt(100);
+        System.out.println("Producer " + randomInt);
+        try {
+            buffer.put(randomInt);
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void consumer() {
+        Integer poll;
+        try {
+            poll = buffer.poll(1, TimeUnit.MILLISECONDS);
```

Comment:
N: we need to use take/put methods that provide "blocking"(wait/notify) API under the hood. Everything is fine.